tests/OTAnalytics/application/test_track_filter.py

```diff
@@ -25,20 +27,34 @@ def track(track_builder: TrackBuilder) -> Track:
     return track_builder.build_track()
 
 
-class TestTrackIsWithinDate:
-    def test_date_within_range(self, track: Track) -> None:
+class TestTrackStartsAtOrAfterDate:
+    def test_starts_at_or_after_date(self, track: Track) -> None:
         start_date = datetime(2000, 1, 1)
-        end_date = datetime(2000, 1, 3)
 
-        predicate = TrackIsWithinDate(start_date, end_date)
+        predicate = TrackStartsAtOrAfterDate(start_date)
         result = predicate.test(track)
         assert result is True
 
     def test_date_outside_range(self, track: Track) -> None:
         start_date = datetime(2000, 1, 5)
-        end_date = datetime(2000, 1, 10)
 
-        predicate = TrackIsWithinDate(start_date, end_date)
+        predicate = TrackStartsAtOrAfterDate(start_date)
+        result = predicate.test(track)
+        assert result is False
```

Comment:
It looks like we are comparing a lot of predicates. Can we refactor them all to one parameterized test?

e.g.
```python
[
    (TrackStartsAtOrAfterDate(datetime(2000,1,1)), True),
    (TrackStartsAtOrAfterDate(datetime(2000,1,5)), False),
    (TrackEndsBeforeOrAtDate(datetime(2000,1,3)), True),
    (TrackEndsBeforeOrAtDate(datetime(2000,1,1)), True),
    (TrackHasClassification(["car", "truck"]), True),
]
````

and so on